src/aggregator/aggregation/gauge.go

```diff
@@ -76,6 +75,12 @@ func (g *Gauge) UpdatePrevious(timestamp time.Time, value float64, prevValue flo
 	g.count--
 	// add the new value to the totals.
 	g.updateTotals(timestamp, value)
+	if math.IsNaN(g.max) || g.max < value {
```

Comment:
i would just update `updateTotals` to include the min/max logic now. then you can also remove it from `Update` and the comment about you can't update min/max :)